lib/s3_meta_sync/syncer.rb

```diff
@@ -244,21 +263,26 @@ def download_content(path)
           "https://s3#{"-#{region}" if region}.amazonaws.com/#{@bucket}/#{path}"
         end
       options = (@config[:ssl_none] ? {:ssl_verify_mode => OpenSSL::SSL::VERIFY_NONE} : {})
-      open(url, options)
-    rescue OpenURI::HTTPError
-      retries ||= 0
-      retries += 1
-      if retries >= 3
+      with_retries(path: path, url: url) { open(url, options) }
+    end
+
+    def with_retries(path:, url:)
+      yield
+    rescue OpenURI::HTTPError, Errno::ECONNRESET => e
+      max_retries = @config[:max_retries] || 2
+      http_error_retries ||= 0
+      http_error_retries += 1
+      if http_error_retries <= max_retries
+        log "#{e.class} error downloading #{path}, retrying #{http_error_retries}/#{max_retries}"
```

Comment:
inconsistent ... either show path or url or both in error message ... see `$!.message <<` below